src/Tizen.Multimedia/AudioManager/SoundEffect.cs

```diff
@@ -0,0 +1,226 @@
+    {
+        ReferenceCopy = 1,
+        AecSpeex = 2,
+        AecWebrtc = 4
+    }
+
+    internal static class EnumExtensions
+    {
+        internal static int ToNative(this SoundEffectType effect)
+        {
+            int ret = 0;
+
+            /* Native enum values are defined as below:
+                typedef enum {
+                    SOUND_EFFECT_REFERENCE_COPY              = 0x0001, //< Including reference source
+                    SOUND_EFFECT_ACOUSTIC_ECHO_CANCEL_SPEEX  = 0x0002, //< Acoustic echo cancel with speex
+                    SOUND_EFFECT_ACOUSTIC_ECHO_CANCEL_WEBRTC = 0x0004, //< Acoustic echo cancel with webrtc
+                } sound_effect_method_with_reference_e;
+
+                typedef enum {
+                    SOUND_EFFECT_NOISE_SUPPRESSION_VOIP          = 0x0001, //< Noise suppression for voice call
+                    SOUND_EFFECT_AUTOMATIC_GAIN_CONTROL_CAPTURE  = 0x0002, //< Auto Gain Control for normal capturing
+                } sound_effect_method_e;
+            */
+            switch (effect)
+            {
+                case SoundEffectType.NoiseSuppression:
+                    ret = 1;
+                    break;
+                case SoundEffectType.AutoGainControl:
+                    ret = 2;
+                    break;
+                case SoundEffectType.NsWithAgc:
+                    ret = 3;
+                    break;
+                case SoundEffectType.ReferenceCopy:
+                    ret = 1;
+                    break;
+                case SoundEffectType.AecSpeex:
+                    ret = 2;
+                    break;
+                case SoundEffectType.AecWebrtc:
+                    ret = 4;
+                    break;
+                default:
+                    break;
+            }
+
+            return ret;
+        }
+
+        internal static SoundEffectType ToPublic(this SoundEffectNative effect)
+        {
+            SoundEffectType ret = SoundEffectType.NoiseSuppression;
+
+            /* Native enum values are defined as below. And these can be set together.
+                typedef enum {
+                    SOUND_EFFECT_NOISE_SUPPRESSION_VOIP          = 0x0001, //< Noise suppression for voice call
+                    SOUND_EFFECT_AUTOMATIC_GAIN_CONTROL_CAPTURE  = 0x0002, //< Auto Gain Control for normal capturing
+                } sound_effect_method_e;
+            */
+            switch (effect)
+            {
+                case SoundEffectNative.NoiseSuppression:
+                    ret = SoundEffectType.NoiseSuppression;
+                    break;
+                case SoundEffectNative.AutoGainControl:
+                    ret = SoundEffectType.AutoGainControl;
+                    break;
+                case SoundEffectNative.NoiseSuppression | SoundEffectNative.AutoGainControl:
+                    ret = SoundEffectType.NsWithAgc;
+                    break;
+                default:
+                    break;
```

Comment:
not an API matter, but you might consider logging an error in such cases

Reply:
I'll add error message.